src/code_b72b0_len_15ed0.c

```diff
@@ -110,22 +372,162 @@ void init_dynamic_entity_list(void) {
+    }
+    newDynEntity->draw = drawFunc;
+    if (drawFunc == NULL) {
+        newDynEntity->draw = &stub_dynamic_entity_delegate;
+    }
+
+    if (gGameStatusPtr->isBattle) {
+        i |= 0x800;
+    }
+    return i;
+}
+
+s32 bind_dynamic_entity_7(void (*updateFunc)(void), void (*drawFunc)(void)) {
+    s32 i;
+    DynamicEntity* newDynEntity;
+
+    for (i = 0; i < MAX_DYNAMIC_ENTITIES; i++) {
+        DynamicEntity* dynEntity = (*gCurrentDynamicEntityListPtr)[i];
+        if (dynEntity == NULL) {
+            break;
+        }
+    }
+    ASSERT(i < MAX_DYNAMIC_ENTITIES);
+
+    (*gCurrentDynamicEntityListPtr)[i] = newDynEntity = heap_malloc(sizeof(DynamicEntity));
+    ASSERT(newDynEntity != NULL);
+
+    newDynEntity->flags = 7;
+    newDynEntity->update = updateFunc;
+    if (updateFunc == NULL) {
+        newDynEntity->update = &stub_dynamic_entity_delegate;
+    }
+    newDynEntity->draw = drawFunc;
+    if (drawFunc == NULL) {
+        newDynEntity->draw = &stub_dynamic_entity_delegate;
+    }
+
+    if (gGameStatusPtr->isBattle) {
+        i |= 0x800;
+    }
+    return i;
+}
+
+s32 bind_dynamic_entity_B(void (*updateFunc)(void), void (*drawFunc)(void)) {
+    s32 i;
+    DynamicEntity* newDynEntity;
+
+    for (i = 0; i < MAX_DYNAMIC_ENTITIES; i++) {
+        DynamicEntity* dynEntity = (*gCurrentDynamicEntityListPtr)[i];
+        if (dynEntity == NULL) {
+            break;
+        }
+    }
+    ASSERT(i < MAX_DYNAMIC_ENTITIES);
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", bind_dynamic_entity_7, s32* updateFunc, void (*drawFunc)(void));
+    (*gCurrentDynamicEntityListPtr)[i] = newDynEntity = heap_malloc(sizeof(DynamicEntity));
+    ASSERT(newDynEntity != NULL);
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", bind_dynamic_entity_B);
+    newDynEntity->flags = 0xB;
+    newDynEntity->update = updateFunc;
+    if (updateFunc == NULL) {
+        newDynEntity->update = &stub_dynamic_entity_delegate;
+    }
+    newDynEntity->draw = drawFunc;
+    if (drawFunc == NULL) {
+        newDynEntity->draw = &stub_dynamic_entity_delegate;
+    }
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", update_dynamic_entities);
+    if (gGameStatusPtr->isBattle) {
+        i |= 0x800;
+    }
+    return i;
+}
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", render_dynamic_entities);
+void update_dynamic_entities(void) {
+    s32 i;
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", func_801234E0);
+    for (i = 0; i < MAX_DYNAMIC_ENTITIES; i++) {
+        DynamicEntity* entity = (*gCurrentDynamicEntityListPtr)[i];
+        if (entity != NULL) {
+            entity->flags &= ~2;
+            entity->update();
+        }
+    }
+}
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", func_80123550);
+void render_dynamic_entities(void) {
+    s32 i;
 
-INCLUDE_ASM(s32, "code_b72b0_len_15ed0", func_801235C0);
+    for (i = 0; i < MAX_DYNAMIC_ENTITIES; i++) {
+        DynamicEntity* entity = (*gCurrentDynamicEntityListPtr)[i];
+        if ((entity != NULL) && !(entity->flags & 2)) {
```

Comment:
Can you remove the parens around `(entity != NULL)` here and the below function? I'd rather not have extraneous parentheses